src/idl_gen_general.cpp

```diff
@@ -409,6 +409,28 @@ static std::string GenDefaultValue(const LanguageParameters &lang, const Value &
            : value.constant;
 }
 
+static std::string GenEnumDefaultValue(const FieldDef &field) {
+  auto enum_def = field.value.type.enum_def;
+  auto vec = enum_def->vals.vec;
+  auto default_value = StringToInt(field.value.constant.c_str());
+
+  // at least, enum has default 0 value.
+  auto result = enum_def->name + "." + (**vec.begin()).name;
+  if (default_value != 0) {
```

Comment:
a union always starts with 0, but an enum may start with any number, so `result` may be incorrect. Instead, remove this `if`, and just return `field.value.constant` as the default value. I believe it should be impossible for your loop not find the value, so this should be ok. An assert/error would be better, but we don't have a mechanism for errors in codegen.